### PR TITLE
Eyal application to employee

### PR DESCRIPTION
**Pull request by pinkasey**
Description of the issue/feature this PR addresses:
When converting an application to an employee, it is reasonable to expect that all custom fields that exist on both - will be copied from the applicant to the newly created employee

Current behavior before PR:
a hard-coded set of fields are copied from the applicant to the new employee (see `create_employee_from_applicant()`).
custom fields are not populated.

Desired behavior after PR is merged:
When the same field is defined on both applicant and employee - it should be copied from applicant to the new employee

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pinkasey** [comment]
I wasn't sure if this should go to 13 or master -
I think the risk here is low (only affect the button "Create Employee"), and the value is much higher than the risk.

If you disagree, I can rebase to master and open a new pull-request.

**Comment by mart-e** [comment]
Hello,

Sorry but I don't think it's a good idea. You are making the assumption that all the custom fields of `hr.applicant` must be copied on the `hr.employee` record.
Either we apply this logic to all fields (custom or not), either we do it case by case.

I would tend for the second case for safety. For instance, we have a field `active` on `hr.applicant` and we do not want to copy it when creating an `hr.employee` record. There is certainly other similar examples with your custom fields. Either I would say that we create a `_prepare_...` method that builds the dict and custom code can customize this.

But maybe @tivisse disagree with me and has another idea.

**Comment by tivisse** [comment]
Hi @pinkasey 

Thanks for the feedback. I tend to agree with  @mart-e .

What if :
- The field ``foo`` on an applicant and an employee don't have the same meaning ? (active is a good example).
- On your test you only take ``Char`` fields into account. But what about ``One2many`` and ``Many2many`` fields ? They're not supposed to be copied by default. Why don't you take the ``ir.model.field`` definition into account ? 
- What about computed/related fields ? What if I have a field on the employee that is related to a field on the applicant ? 
- What about non stored fields with an inverse method ? They could erase/modify a value that you wanted to copy. What about ``Reference`` fields?

**Comment by pinkasey** [comment]
Thanks for the feedback, @mart-e  @tivisse , I appreciate it.
TL;DR - 
it seems to me that I need to play with this feature some more, and propose it again when it's more mature.

I want to respond briefly to your comments (I started writing a response and it got too long, ):
1. why just custom fields - because I don't want to risk messing with non-custom fields. and I know for a fact that some should not be copied (e.g `name`, `active`), I don't know what else I'm missing. I realise 
2. custom-fields might also have different meaning - I think it's less risky. I think a good mitigation would be to put this feature behind a configuration, turned off by default.
3. > create a _prepare_... method that builds the dict and custom code can customize this.
  
     I think a feature like that should be configurable without code. I we make it customizable, it's best if we have a nice UI for it, with validation and all. but I realize that's a lot of work.
4. > Why don't you take the ir.model.field definition into account ?

      you're right, I should. I'll do that.
5. computed/related/stored fields etc - I should filter those and not copy them. you're right.

**Comment by pinkasey** [comment]
I'm truly sorry for wasting your time.
I should play with this feature in my own org, and hopefully come up with a better PR.

I think your approach is better, so at least you got me going in the right direction.

**Comment by tivisse** [comment]
No harm done ! Thanks for your feedback anyway ;)
